UnitTests/UI/Test_AdvancedCollectionView.cs

```diff
@@ -70,6 +70,117 @@ public void Test_AdvancedCollectionView_Filter()
 
         [TestCategory("AdvancedCollectionView")]
         [UITestMethod]
+        public void Test_AdvancedCollectionView_FilterOrder()
+        {
+            var l = new ObservableCollection<string>
+            {
+                "lorem",
+                "ipsum",
+                "dolor",
+                "sit",
+                "amet"
+            };
+
+            var a = new AdvancedCollectionView(l)
+            {
+                Filter = (x) => x.ToString().Length < 5
+            };
+
```

Comment:
You can add asserts here before the insert to make sure the filter before insert is in the correct state as well.  Same with some of the other copies of this below.

Reply:
Done 😃